pxd.c

```diff
@@ -1040,8 +1042,10 @@ static blk_status_t pxd_queue_rq(struct blk_mq_hw_ctx *hctx,
 	struct request *rq = bd->rq;
 	struct pxd_device *pxd_dev = rq->q->queuedata;
 	struct fuse_req *req = blk_mq_rq_to_pdu(rq);
+	struct fuse_conn *fc = &pxd_dev->ctx->fc;
 
-	if (BLK_RQ_IS_PASSTHROUGH(rq))
+	if (BLK_RQ_IS_PASSTHROUGH(rq) ||
+		(!READ_ONCE(fc->allow_disconnected) && !READ_ONCE(fc->connected)))
```

Comment:
Is the duplicate check necessary?

Reply:
fc->connected can be killed, checking just for allow_disconnected is sufficient.